### ISSUE TITLE
BeaconInitProvider.onCreate() causes crashes on pre lollipop

### ISSUE THREAD

**Issue by taaeng** [helpscout/beacon-android-sdk-sample#40]
Since the minimum supported SDK is 21, I don't run HelpScout code on old devices.
But still, the app crashes on startup.
Can you avoid doing this initial setup on pre SDK 21?

`    java.lang.NoClassDefFoundError: com.jakewharton.a.b
        at com.jakewharton.a.a.a(AndroidThreeTen.java:18)
        at com.helpscout.beacon.BeaconInitProvider.onCreate(BeaconInitProvider.kt:19)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1214)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1189)
        at android.app.ActivityThread.installProvider(ActivityThread.java:4879)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:4485)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4425)
        at android.app.ActivityThread.access$1300(ActivityThread.java:141)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1316)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:137)
        at android.app.ActivityThread.main(ActivityThread.java:5103)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:525)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
        at dalvik.system.NativeStart.main(Native Method)`

**Comment by scottyab** [comment]
Hey @taaeng thanks for flagging this. I assume you're using `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)` and if true Building the Beacon? 

There's a couple of ways we could solve this. By either not loading the `BeaconInitProvider` or an `if` check similar to above in  `BeaconInitProvider.onCreate()` 

I'll add this issue to our tracker.

**Comment by taaeng** [comment]
Thanks.
The thing is; there is no code for me to wrap in an if(). As of now, all code is commented out ... The only thing I'm doing is adding the beacon libs to my build gradle file, and that causes the Kikat crash..  To me it seems like the BeaconInitProvider initializes itself. 🤔

**Comment by scottyab** [comment]
Hey @taaeng fix coming in the next release. That disables this auto initialisation for devices running below Android 5.0. Thanks

**Comment by scottyab** [comment]
This is resolved in v0.3.1. The `BeaconInitProvider` now will not initialise if the Android version is below 5.0. We've also added a way to stop the auto-init via a meta-data flag. Please note if you do wish to stop the auto-init you'll need to manually init by passing an application context to `Beacon.Builder.withContxt(..)`
```
<meta-data android:name="com.helpscout.beacon.SDKAutoInitialise" android:value="false" />
```

Thanks for rasing the issue.
